lib/googleauth/credentials_loader.rb

```diff
@@ -50,10 +50,11 @@ module CredentialsLoader
                             "s.googleusercontent.com".freeze
 
       CLOUD_SDK_CREDENTIALS_WARNING =
-        "Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most " \
-        "server applications use service accounts instead. If your application continues to use end user credentials " \
-        'from Cloud SDK, you might receive a "quota exceeded" or "API not enabled" error. For more information about ' \
-        "service accounts, see https://cloud.google.com/docs/authentication/. To suppress this message, set the " \
+        "You are authenticating using user credentials." \
```

Comment:
Per the document we want to REMOVe the old message. (1.2.1)
The substitution has different logic (show in prod only) and for now we implement it only for Java/Python. (1.2.2)

So in case of Ruby you can just remove this logic and message completely.